data_animation.py

```diff
@@ -0,0 +1,269 @@
+                self.time_reference[self.time_cursor] <= self.time):
+
+            # Add a recording frame to the ploted time, then to each value
+            self.plotted_time.append(self.time_reference[self.time_cursor])
+
+            for line in self.lines:
+                line.tick(self.time_cursor, self.plotted_time)
+
+            if self.time_cursor < len(self.time_reference):
+                self.time_cursor += 1
+
+        updated_lines = [line.line for line in self.lines]
+
+        if frame_number == 0:
+            self.prev_timer_value = perf_counter()
+
+        ## Log the render progress
+        if (frame_number + 1) % self.LOGGING_INTERVAL == 0:
+
+            # add 1 to frame counts to adjust for 0 index
+            percent_completed = round((frame_number + 1)/self.config_ani.FRAME_LENGTH*100, 2)
+            print_message_frames = f"Rendering frame {frame_number + 1}" +\
+                    f" of {self.config_ani.FRAME_LENGTH} ({percent_completed}%)"
+
+            ## Handle a non-cringers way of estimating remaining time
+            if self.prev_timer_value == -1:
+                self.prev_timer_value = perf_counter()
+                print_message_timing = "Preparing time remaining estimate..."
+                print_message_estimate = ""
+            else:
+                timer_duration = (perf_counter() - self.prev_timer_value) # value in seconds
+                percent_remaining = 100.0 - percent_completed
+
+                remaining_time_estimate = timer_duration*(percent_remaining/self.INTERVAL_PERCENT)
+                print_message_timing = f"Time to render frame batch: {round(timer_duration, 2)} s"
+                print_message_estimate = \
+                    f"Estimated time remaining: {round(remaining_time_estimate, 2)} s"
+
+
+                self.prev_timer_value = perf_counter() # reset timer
+
+            print_message = print_message_frames + '  |  ' + print_message_timing
+            print_message = print_message + '  |  ' + print_message_estimate
+
+            print_message = print_message + " "*10 # For clearing stray chars by carriage return
+
+            print(print_message, end="\r")
+
+        return updated_lines
+
+
+    def create_animation(self):
+        """Main method for creating an animation."""
+
+
+        plt.rcParams["animation.ffmpeg_path"] = self.config_setup.FFMPEG_PATH
+        m = pd.read_csv(os.path.join(self.config_setup.DATA_WORKING_DIR, \
+                                     self.config_ani.DATA_FILE_NAME))
+
+        # Set the dimensions of the plot
+        plt.rcParams["figure.figsize"] = (self.config_ani.PLOT_WIDTH, self.config_ani.PLOT_HEIGHT)
+
+        # Timestap as a base refference
+        self.time_reference = m["Timestamp"]
+        self.plotted_time = []
+        self.time_cursor = 0
+        self.time = self.config_ani.STARTING_TIME  # animiation progress in seconds
+
+        self.fig, _ = plt.subplots()
```

Comment:
Better practice to use `self.fig = plt.subplots()[0]`. `_` kills readability especially if it's used a lot

Reply:
done